validation/checks/license_check.go

```diff
@@ -0,0 +1,84 @@
+package checks
+
+import (
+	"io"
+	"net/http"
+	"regexp"
+
+        "github.com/UCLALibrary/validation-service/validation/config"
+        "github.com/UCLALibrary/validation-service/validation/csv"
+)
+
+// Error messages
+var (
+	noProfileErr = "supplied profile cannot be nil"
+	urlFormatErr = "license URL is not in a proper format (check for HTTPS)"
+	urlConnectErr = "problem connecting to license URL"
+	urlReadErr = "problem reading body of license URL"
+)
+
+type LicenseCheck struct{
+	profiles *config.Profiles
+}
+
+func (check *LicenseCheck) NewLicenseLCheck(profiles *config.Profiles) (*LicenseCheck, error) {
+	if profiles == nil {
+		return nil, csv.NewError(noProfileErr, csv.Location{}, "nil")
+	}
+
+	return &LicenseCheck{
+		profiles: profiles,
+	}, nil
+}
+
+func (check *LicenseCheck) Validate(profile string, location csv.Location, csvData [][]string) error {
+	// license not relevant to Bucketeer processing
+	if profile == "bucketeer" {
+		return nil
+	}
+
+	if err := csv.IsValidLocation(location, csvData, profile); err != nil {
+		return err
+	}
+
+	// find the header and determine if it matches an ark header
```

Comment:
Also, I think this comment should be "...matches a License header" instead of ark header(?)

Reply:
D'oh! Cut-and-paste brain.